### DISCUSSION TITLE
Per-channel context history?

### DISCUSSION THREAD

**Discussion by mflux** [Zero6992/chatGPT-discord-bot#512] [General]
I'm trying to build a bot that can create a narrative roleplaying game. However I noticed that the bot keeps the same context-history regardless of channel. Is there some setting that I could toggle to cause it to scope the context to specific channels?

Or is this a feature that could be added in the future? 

Thanks for the great project.

**Comment by mflux** [comment]
In addition, is there some configuration (or let me know where to begin looking to modify) letting the bot know who is speaking? Currently it thinks all chat originates from the same user.

**Reply by sayasukasusu** [reply]
I edited the async def send_message function in src/aclient.py to identify the user
```
async def send_message(self, message, user_message):
        if self.is_replying_all == "False":
            author = message.user.id
        else:
            author = message.author.id
        try:
            username = await discord.Client.fetch_user(self, author)
            try:
                response = await self.handle_response(user_message + " I am " + username.global_name)
            except:
                response = await self.handle_response(user_message + " I am " + username.name)

            response_content = f'> **{user_message[:300]}** - <@{str(author)}> \n\n{response}'
            await send_split_message(self, response_content, message)
        except Exception as e:
            logger.exception(f"Error while sending : {e}")
            # Error handling as before

```
